src/main/java/com/google/sps/servlets/TripServlet.java

```diff
@@ -18,16 +33,155 @@
+      this.tripDayOfTravel, this.photoSrc);
+
+    /**
+     * TODO: Remaining code for storing Event and TripDay objects should 
+     * go here, below the above code, as the Trip has to be set first in order
+     * to maintain Entity hierarchy / ancestor paths. 
+     * 
+     * Below methods can also use the field variables fetched from request in 
+     * the above code.
+     */
+
+    // Redirect to the "/trips/" page to show the trip that was added.
+    response.sendRedirect("/trips/");
+  }
+
+  /**
+   * Get the place ID of the text search. Return null if no place ID matches
+   * the search.
+   */ 
+  public String getPlaceIdFromTextSearch(GeoApiContext context, String textSearch) 
+    throws IOException {
+
+    FindPlaceFromTextRequest findPlaceRequest = PlacesApi.findPlaceFromText(context, 
+      textSearch, FindPlaceFromTextRequest.InputType.TEXT_QUERY);
+
+    try {
+      FindPlaceFromText findPlaceResult = findPlaceRequest.await();
+
+      // Return place ID of the first candidate result.
+      if (findPlaceResult.candidates != null) {
+        return findPlaceResult.candidates[0].placeId;
+      }
+      
+      // No candidate is given, so return null.
+      return null;
+    } catch(ApiException | InterruptedException e) {
+      throw new IOException(e);
     }
   }
 
+  /**
+   * Get the PlaceDetails object from the place ID.
+   */
+  public PlaceDetails getPlaceDetailsFromPlaceId(GeoApiContext context, String placeId)
+    throws IOException {
+
+    PlaceDetailsRequest placeDetailsRequest = PlacesApi.placeDetails(context, 
+      placeId);
+    try {
+      return placeDetailsRequest.await();
+    } catch(ApiException | InterruptedException e) {
+      throw new IOException(e);
+    }
+  }
+
+  /**
+   * Populate the destinationName and photoSrc fields using the Google Maps API.
+   */
+  public void populateDestinationAndPhoto(GeoApiContext context, String tripDestination)
+    throws IOException {
+
+    // Get place ID from search of trip destination. Get photo and destination 
+    // if not null; otherwise, use a placeholder photo and destination.
+    String destinationPlaceId = getPlaceIdFromTextSearch(context, this.tripDestination);
+    if (destinationPlaceId == null) {
+      this.destinationName = tripDestination;
+      this.photoSrc = "../images/placeholder_image.png";
+    } else {
+      PlaceDetails placeDetailsResult = getPlaceDetailsFromPlaceId(context, destinationPlaceId);
+
+      // Get the name of the location from the place details result.
+      this.destinationName = placeDetailsResult.name;
+
+      // Get a photo of the location from the place details result.
+      if (placeDetailsResult.photos == null) {
+        this.photoSrc = "../images/placeholder_image.png";
+      } else {
+        Photo photoObject = placeDetailsResult.photos[0];
+        this.photoSrc = getUrlFromPhotoReference(400, photoObject.photoReference);
```

Comment:
nit: save 400 as a constant variable (for easier change later on)

Reply:
That's smart, I'll do that! Thanks 👍